tools/moth.py

```diff
@@ -328,15 +334,35 @@ def __init__(self, path, seed):
         self.path = path
         self.seed = seed
         self.pointvals = []
+        self.catmod = None
+
+        try:
+            catmod = SourceFileLoader(
+                'catmod',
+                os.path.join(path, 'category.py')).load_module()
+            assert all([
+                hasattr(catmod, 'make'),
+                hasattr(catmod, 'points'),
+                type(catmod.points) is list,
+            ])
+            self.catmod = catmod
+            self.pointvals.extend(catmod.points)
+        except:
+            pass
+
         for fpath in glob.glob(os.path.join(path, "[0-9]*")):
             pn = os.path.basename(fpath)
             points = int(pn)
             self.pointvals.append(points)
+
         self.pointvals.sort()
 
     def puzzle(self, points):
-        path = os.path.join(self.path, str(points))
-        return Puzzle(self.seed, path)
+        if self.catmod is not None and points in self.catmod.points:
+            return Puzzle(self.seed, points=points, category=self.catmod)
```

Comment:
toe-may-toe/toe-mah-toe ... The initializer for Puzzle is fairly strict and needs a signal that it is to be dynamically made. Passing the category-maker instance is the signal. We could revert to a PuzzleFactory pattern and make Puzzle a smart dict with no instantiation as well.